server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/builder/OffHeapStarTreeBuilderTests.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.builder;
+
+import org.apache.lucene.index.SegmentWriteState;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeField;
+import org.opensearch.index.mapper.MapperService;
+
+import java.io.IOException;
+
+public class OffHeapStarTreeBuilderTests extends AbstractStarTreeBuilderTests {
+    @Override
+    public BaseStarTreeBuilder getStarTreeBuilder(
```

Comment:
Lets also look into how we can introduce some randomness in data generation and evaluate pre-computed aggregations against a brute-force logic in our tests in addition to current pre-determined data sets

Reply:
Agreed. https://github.com/opensearch-project/OpenSearch/issues/14813#issuecomment-2265008905